examples/jsm/webxr/ARButton.js

```diff
@@ -1,40 +1,38 @@
-var ARButton = {
+class ARButton {
 
-	createButton: function ( renderer, sessionInit = {} ) {
+	static createButton( renderer, sessionInit = {} ) {
 
-		function showStartAR( /*device*/ ) {
+		let currentSession = null;
 
-			var currentSession = null;
+		const button = document.createElement( 'button' );
+		const message = document.createElement( 'a' );
 
-			function onSessionStarted( session ) {
+		function onSessionEnded( /*event*/ ) {
```

Comment:
More questions... Why did you change the order of the functions?

Reply:
eslint complains and suggest to declare functions before they are used, so they are all now declared before they have been called